MongoDB.Entities/Core/Cache.cs

```diff
@@ -158,6 +158,7 @@ internal static IEnumerable<PropertyInfo> UpdatableProps(T entity)
     {
         var parameter = Expression.Parameter(typeof(T), "t");
         var property = Expression.Property(parameter, idProp);
-        return Expression.Lambda<Func<T, object?>>(property, parameter);
+        Expression conversion = Expression.Convert(property, typeof(object));

```

Comment:
looks like this conversion expression is not supported by mongo driver :-(

![screen-record-1679888365](https://user-images.githubusercontent.com/7043768/227835361-8fce8f44-3369-43f5-ba9e-1c2a4e90de1d.png)